dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/TemplatePathStrategyResolver.java

```diff
@@ -137,31 +138,31 @@ private boolean isPath (final String templatePath) {
         }
 
         private String getPath(final RenderParams params, final String path, final String uid) {
-
-
-            final Host host     = params.currentHost;
-            String templatePath = path;
-
-
             try {
-
-                templatePath =(!templatePath.startsWith(HOST_INDICATOR))?
-                        builder(FORWARD_SLASH, params.mode.name(), FORWARD_SLASH, HOST_INDICATOR,
-                            this.getHost(host).getHostname(), path.startsWith(FORWARD_SLASH)? StringPool.BLANK:FORWARD_SLASH,
-                            path, FORWARD_SLASH, uid, PERIOD, VelocityType.CONTAINER.fileExtension).toString():
-
-                        builder(FORWARD_SLASH, params.mode.name(), FORWARD_SLASH,
-                            path, FORWARD_SLASH, uid, PERIOD, VelocityType.CONTAINER.fileExtension).toString();
+                return FileAssetContainerUtil.getInstance().isFullPath(path) ?
+                        getContainerResourcePathFromFullPath(params, path, uid) :
+                        getContainerResourceFromRelativePath(params, path, uid);
             } catch (Exception e) {
 
-                Logger.warn(this.getClass(), " - unable to resolve " + templatePath + " getting this: "+ e.getMessage() );
+                Logger.warn(this.getClass(), " - unable to resolve " + path + " getting this: "+ e.getMessage() );
                 if(e.getStackTrace().length>0) {
                     Logger.warn(this.getClass(), " - at " + e.getStackTrace()[0]);
                 }
                 throw new DotStateException(e);
             }
+        }
+
+        @NotNull
+        private String getContainerResourceFromRelativePath(RenderParams params, String path, String uid) {
+            return builder(FORWARD_SLASH, params.mode.name(), FORWARD_SLASH, HOST_INDICATOR,
+                    params.currentHost.getHostname(), path.startsWith(FORWARD_SLASH)? StringPool.BLANK:FORWARD_SLASH,
+                    path, FORWARD_SLASH, uid, PERIOD, VelocityType.CONTAINER.fileExtension).toString();
+        }
 
-            return templatePath;
+        @NotNull
+        private String getContainerResourcePathFromFullPath(RenderParams params, String path, String uid) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'uid' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5067852)